pandas/tests/indexes/test_datetimelike.py

```diff
@@ -732,6 +741,13 @@ def test_fillna_datetime64(self):
                            dtype=object)
             self.assert_index_equal(idx.fillna('x'), exp)
 
+    def test_difference_freq(self):
+        # GH14323. difference of DatetimeIndex does not preserve frequency,
+        # so a differencing operation should not retain the freq field of the
+        # original index.
+        i = self.create_index()
+        self.assertIsNone(i.difference(i[1:3]).freq)
```

Comment:
Sorry for the wait-- took me a bit to get the git to work (my own confusion). I've reverted the whatsnew changes and changed the tests to use explicit comparison indexes, at the expense of subclassing. 

@jorisvandenbossche Happy to refactor the tests again to use subclassing if it's helpful, I'm just not sure I can do that _and_ make the desired result explicit. From what I can tell, the way self.indices is constructed all of them correctly have freq <Day>, so I can't use them as a desired result. Conversely if want to explicitly create a comparison index, I need to know what class I'm working with, since TimedeltaIndex and DatetimeIndex don't take the same set of string inputs.